src/v/cloud_storage/topic_manifest.cc

```diff
@@ -172,6 +173,13 @@ struct topic_manifest_handler
 
     bool Null() {
         if (_state == state::expect_value) {
+            if (_key == "retention_bytes") {
+                _properties.retention_bytes = tristate<size_t>{std::nullopt};
+            } else if (_key == "retention_duration") {
+                _properties.retention_duration
+                  = tristate<std::chrono::milliseconds>{std::nullopt};
```

Comment:
Do we have tests for this case?

Reply:
Fell on the cutting floor, I have it partially here https://github.com/redpanda-data/redpanda/commit/d8b22a2c46d988d1e84d5e3562f912decc19a78b#diff-f49d3c57c7482a16b5a65be7f49f0f5bac199e27458e221f0c404d410b08b4b0L58-R60

and you can try that this works 
```
diff --git a/src/v/cloud_storage/tests/topic_manifest_test.cc b/src/v/cloud_storage/tests/topic_manifest_test.cc
index e926c274a2..0ff3586876 100644
--- a/src/v/cloud_storage/tests/topic_manifest_test.cc
+++ b/src/v/cloud_storage/tests/topic_manifest_test.cc
@@ -55,7 +55,9 @@ static constexpr std::string_view min_topic_manifest_json = R"json({
     "cleanup_policy_bitflags": null,
     "compaction_strategy": null,
     "timestamp_type": null,
-    "segment_size": null
+    "segment_size": null,
+    "retention_bytes": null,
+    "retention_duration": null
 })json";
 
 static constexpr std::string_view full_topic_manifest_json = R"json({
@@ -205,8 +207,8 @@ SEASTAR_THREAD_TEST_CASE(min_config_update_all_fields_correct) {
     BOOST_REQUIRE(!topic_config.properties.compaction_strategy);
     BOOST_REQUIRE(!topic_config.properties.timestamp_type);
     BOOST_REQUIRE(!topic_config.properties.segment_size);
-    BOOST_REQUIRE(topic_config.properties.retention_bytes.is_disabled());
-    BOOST_REQUIRE(topic_config.properties.retention_duration.is_disabled());
+    BOOST_REQUIRE(topic_config.properties.retention_bytes.is_empty());
+    BOOST_REQUIRE(topic_config.properties.retention_duration.is_empty());
 }
 
 SEASTAR_THREAD_TEST_CASE(full_config_update_all_fields_correct) {
```

i can add it in the followup or here, if i have to touch other stuff

Reply:
Got it, thanks! Yeah fine with me either way as long as we know this works